src/System/refine.rs

```diff
@@ -330,7 +331,9 @@ fn build_state_pair<'a>(
     ) {
         new_sp_zone = test_zone2;
     }
+
     new_sp.zone = new_sp_zone;
+    new_sp.calculate_max_bound(sys1, sys2, is_state1);
 
     if is_new_state(&mut new_sp, passed_list) && is_new_state(&mut new_sp, waiting_list) {
```

Comment:
Right before this check we should apply the extrapolation to the state zone (rs_dbm_extrapolateMaxBounds)